### PR TITLE
B37 gearshift

### PR DESCRIPTION
**Pull request by RoanKanninga**

### CODE REVIEW

File: generate_dragenScripts.sh [hunk 1]
```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+if module list | grep -o -P 'NGS_DNA(.+)'
```
Comment (pneerincx) [inline generate_dragenScripts.sh:2]:
```suggestion
if module list -t NGS_DNA | grep NGS_DNA 
```

File: generate_dragenScripts.sh [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+#
+python "${EBROOTNGS_DNA}/scripts/sampleSheetChecker.py" "${samplesheet}"
+if [ -f "${samplesheet}.temp" ]
+then
+	mv "${samplesheet}.temp" "${samplesheet}"
+fi
+## adding columns if they are not present in the samplesheet
+for i in "Gender" "MotherSampleId" "FatherSampleId" "MotherAffected" "FatherAffected" "FirstPriority"
+do
+	python "${EBROOTNGS_DNA}/scripts/updatingColumns.py" "${samplesheet}" "${i}" ; mv "${samplesheet}.tmp" "${samplesheet}"
```
Comment (pneerincx) [inline generate_dragenScripts.sh:67]:
```suggestion
	python "${EBROOTNGS_DNA}/scripts/updatingColumns.py" "${samplesheet}" "${i}"
	mv "${samplesheet}.tmp" "${samplesheet}"
```

File: generate_dragenScripts.sh (hunk 1 above)
Comment (RoanKanninga) [reply generate_dragenScripts.sh:2]: your suggestion is not giving the same results:
```
module list -t NGS_Demultiplexing | grep NGS_Demultiplexing
1) NGS_Demultiplexing/2.3.1
```
vs

```
module list | grep -o -P 'NGS_Demultiplexing(.+)'
NGS_Demultiplexing/2.3.1
```

File: generate_dragenScripts.sh [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+echo "tmpName,${tmpDirectory}" > ${genScripts}/tmpdir_parameters.csv
+perl "${EBROOTNGS_DNA}/scripts/convertParametersGitToMolgenis.pl" "${genScripts}/tmpdir_parameters.csv" > "${genScripts}/parameters_tmpdir_converted.csv"
+perl "${EBROOTNGS_DNA}/scripts/convertParametersGitToMolgenis.pl" "${EBROOTNGS_DNA}/parameters.csv" > "${genScripts}/parameters_converted.csv"
+#perl "${EBROOTNGS_DNA}/scripts/convertParametersGitToMolgenis.pl" "${EBROOTNGS_DNA}/parameters_${group}.csv" > "${genScripts}/parameters_group_converted.csv"
+perl "${EBROOTNGS_DNA}/scripts/convertParametersGitToMolgenis.pl" "${EBROOTNGS_DNA}/${environmentParameters}.csv" > "${genScripts}/parameters_environment_converted.csv"
+
+## has to be set, otherwise it will crash due to parameters which are not set, this variable will be updated in the next step
+batching="_small"
+
+ngsversion=$(module list | grep -o -P 'NGS_DNA(.+)')
```
[the remark this answers is not in the document]
Comment (RoanKanninga) [reply generate_dragenScripts.sh:93]: same comment as before

[the remark this answers is not in the document]
Comment (pneerincx) [reply generate_dragenScripts.sh:93]: Sorry, `-t` was in the wrong order; option must be added before `list` sub-command:
`ngsversion=$(module -t list NGS_DNA)`
I have updated the suggestion.

File: generate_dragenScripts.sh (hunk 1 above)
Comment (pneerincx) [reply generate_dragenScripts.sh:2]: Sorry, `-t` was in the wrong order; option must be added before `list` sub-command.I have updated the suggestion.

Comment (pneerincx) [reply generate_dragenScripts.sh]: You dropped the `grep` here, but `grep` is required to exit with non zero exit code if NGS_DNA was not loaded. `module list` will not fail no matter whether NGS_DNA is loaded or not.

Comment (RoanKanninga) [reply generate_dragenScripts.sh]: correct, I was too quick

**Review by pneerincx — APPROVED** [review]

**Pull request merged**